qutip/core/qobj.py

```diff
@@ -936,24 +936,23 @@ def sqrtm(self, sparse=False, tol=0, maxiter=100000):
         """
         if self.dims[0] != self.dims[1]:
             raise TypeError('sqrt only valid on square matrices')
-        # TODO: consider another way of handling the dispatch here.
-        if sparse:
-            evals, evecs = _data.eigs_csr(_data.to(_data.CSR, self.data),
+        if isinstance(self.data, _data.CSR) and sparse and self.shape[0] > 3:
```

Comment:
`eigs_csr` should handle the `shape[0] < 3` case itself (using `eigs_dense` if needed).